crates/subspace-node/src/commands/run/consensus.rs

```diff
@@ -161,6 +163,121 @@ struct DsnOptions {
+        match input {
+            "archive" => Ok(Self::Archive),
+            "archive-canonical" => Ok(Self::ArchiveCanonical),
+            _ => Err("Invalid state pruning mode specified".to_string()),
+        }
+    }
+}
+
+impl fmt::Display for StatePruningMode {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str(match self {
+            Self::Archive => "archive",
+            Self::ArchiveCanonical => "archive-canonical",
+        })
+    }
+}
+
+/// This mode specifies when the block's body (including justifications) should be pruned (ie,
+/// removed) from the database.
+#[derive(Debug, Clone, Copy, PartialEq)]
+enum BlocksPruningMode {
+    /// Keep the data of all blocks.
+    Archive,
+    /// Keep only the data of finalized blocks.
+    ArchiveCanonical,
+    /// Keep the data of the last number of finalized blocks.
+    Number(u32),
+}
+
+impl FromStr for BlocksPruningMode {
+    type Err = String;
+
+    fn from_str(input: &str) -> Result<Self, Self::Err> {
+        match input {
+            "archive" => Ok(Self::Archive),
+            "archive-canonical" => Ok(Self::ArchiveCanonical),
+            n => n
+                .parse()
+                .map_err(|_| "Invalid block pruning mode specified".to_string())
+                .map(Self::Number),
+        }
+    }
+}
+
+impl fmt::Display for BlocksPruningMode {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Archive => f.write_str("archive"),
+            Self::ArchiveCanonical => f.write_str("archive-canonical"),
+            Self::Number(n) => f.write_str(n.to_string().as_str()),
+        }
+    }
+}
+
+/// Parameters to define the pruning mode
+#[derive(Debug, Clone, Parser)]
+struct PruningOptions {
+    /// Specify the state pruning mode.
+    ///
+    /// This mode specifies when the block's state (ie, storage) should be pruned (ie, removed)
+    /// from the database.
+    /// This setting can only be set on the first creation of the database. Every subsequent run
+    /// will load the pruning mode from the database and will error if the stored mode doesn't
+    /// match this CLI value. It is fine to drop this CLI flag for subsequent runs.
+    /// Possible values:
+    ///  - archive: Keep the state of all blocks.
+    ///  - archive-canonical: Keep only the state of finalized blocks.
+    #[arg(long, default_value_t = StatePruningMode::ArchiveCanonical)]
+    state_pruning: StatePruningMode,
+
+    /// Specify the blocks pruning mode.
+    ///
+    /// This mode specifies when the block's body (including justifications)
+    /// should be pruned (ie, removed) from the database.
+    /// Possible values:
+    ///  - archive Keep all blocks.
+    ///  - archive-canonical Keep only finalized blocks.
+    ///  - number: Keep the last `number` of finalized blocks.
+    #[arg(long, default_value_t = BlocksPruningMode::Number(256))]
+    blocks_pruning: BlocksPruningMode,
```

Comment:
Shouldn't we also include a check to not let user specify < 256 for some reason or is that not a requirement ?

Reply:
Not a requirement, it is a somewhat arbitrary number